managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedLedgerException.java

```diff
@@ -29,46 +32,61 @@ public ManagedLedgerException(String msg) {
         super(msg);
     }
 
-    public ManagedLedgerException(Throwable e) {
+    protected ManagedLedgerException(Throwable e) {
         super(e);
     }
 
     public ManagedLedgerException(String msg, Throwable e) {
         super(msg, e);
     }
 
-    public static ManagedLedgerException getManagedLedgerException(Throwable e) {
-        if (e instanceof ManagedLedgerException) {
-            return (ManagedLedgerException) e;
+    /**
+     * Returns the encoding of the given (non-null) exception as a wrapped ManagedLedgerException
+     * unless it is one already. Since we use a lot of CompletableFuture, it will unwrap the completion
+     * exception to help get to the original exception.
+     */
+    public static ManagedLedgerException getManagedLedgerException(@Nonnull Throwable ex) {
```

Comment:
what about making this method generic in `FutureUtil` ? 

maybe something like
```
@SneakyThrows
    public static <T extends Throwable> T wrapException(Class<T> toClass, Throwable ex) {
        final Throwable rc = FutureUtil.unwrapCompletionException(ex);
        if (rc.getClass().isAssignableFrom(toClass)) {
            return (T) rc;
        }
        return toClass.getConstructor(Throwable.class).newInstance(rc);
    }
```

Reply:
`return toClass.getConstructor(Throwable.class).newInstance(rc);`
This implies that it works only with Exception types that has that constructor.